shared/src/main/scala/mlscript/helpers.scala

```diff
@@ -6,6 +6,55 @@ import scala.collection.immutable.SortedSet
 
 import mlscript.utils._, shorthands._
 
+/**
+  * A class to hold to the context while converting mlscript types to
+  * typescript representation.
+  * 
+  * polarity tracks polarity of the current type.
+  * * a type has no polarity by default
+  * * a type has negative polarity at function input position
+  * * a type has positive polarity at function output position
```

Comment:
That's actually not how polarities work. For a given definition's type, we start at positive polarity. Function parameters don't turn the polarity negative, but instead invert it. Results leave the polarity unchanged.